x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceAction.java

```diff
@@ -173,6 +173,18 @@ public ActionRequestValidationException validate() {
                 e.addValidationError("input array is empty");
                 return e;
             }
+            if (taskType.equals(TaskType.RERANK)) {
+                if (query == null) {
+                    var e = new ActionRequestValidationException();
+                    e.addValidationError("missing query");
+                    return e;
+                }
+                if (query.isEmpty()) {
+                    var e = new ActionRequestValidationException();
+                    e.addValidationError("query is empty");
```

Comment:
e.addValidationError(format("Field [query] cannot be empty for task type [%s]", TaskType.RERANK));